uswds/uswds
# USWDS - File Input: Move imageId for larger scope. Resolves #5088

## Summary
This change declares `imageId` outside of the `onloadstart()` and `onloadend()` functions to ensure the same target is selected and replaced by the preview image if the attachment takes too long to attach.

## Related issue
Closes #5088 

## Preview link
Preview link: [File Input →](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/cm-file-input-imageID/iframe.html?args=&id=components-form-inputs-file-input--default&viewMode=story)

## Problem statement
> If a file takes too long to attach to the file input, it can cause the loading spinner to not be replaced by the preview image.

`createUniqueID()` uses the current time to create the `ImageId` variable. When a large file takes time to upload, it can cause `onloadstart` and `onloadend` to generate different image IDs. 

## Solution
By declaring the variable before the functions,  we can use `createUniqueID()` only once and pass the variable through both functions.

## Testing and review
1. Upload larger files to file input  _// (I had to use a 600mb dmg file to replicate the error)_
2. Confirm that the spinner is replaced with preview image
3. Make multiple attempts and try file input variants to confirm it still works as expected
